File: lib/camera.py
from vec3 import vec3


class Camera:
    def __init__(self, pos=vec3(0, 0, 0)):
        self.pos = pos

    def log(self):
        print("<Camera pos=[%s, %s, %s]>" %
              (self.pos.x, self.pos.y, self.pos.z))

File: src/camera.rs
extern crate cgmath;

use cgmath::{Vector3};


pub struct Camera {
    pos: Vector3<f32>,
    dir: Vector3<f32>
}

impl Camera {
    pub fn new (px: f32, py: f32, pz: f32) -> Camera {
        Camera {
            pos: Vector3 { x: px, y: py, z: pz },
            dir: Vector3 { x: 0f32, y: 0f32, z: -1f32 }
        }
    }
}
